# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-xgixp/hotfix-lswap.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- xf86-video-xgixp/src/xgi_regs.h.vanilla	2024-04-07 13:36:42.206985723 +0200
+++ xf86-video-xgixp/src/xgi_regs.h	2024-04-07 13:40:26.736984738 +0200
@@ -29,10 +29,11 @@
 #define _XGI_REGS_H_
 
 #include "compiler.h"
+#include <byteswap.h>
 
 #if X_BYTE_ORDER == X_BIG_ENDIAN
-# define BE_SWAP32(v) (lswapl(v))
-# define BE_SWAP16(v) (lswaps(v))
+# define BE_SWAP32(v) (bswap_32(v))
+# define BE_SWAP16(v) (bswap_16(v))
 #else
 # define BE_SWAP32(v) (v)
 # define BE_SWAP16(v) (v)
